src/libretro.c

```diff
@@ -0,0 +1,843 @@
+#include "libretro.h"
+
+#include <stdbool.h>
+#include <time.h>
+#include <string.h>
+#include <math.h>
+
+#include "shared.h"
+#include "sb_types.h"
+#include "gb.h"
+#include "gba.h"
+#include "nds.h"
+
+#define SKYEMU_LIBRETRO_VERSION "0.1.0"
+
+// necessary to resolve a linker error when loaded by retroarch
+bool se_load_bios_file(const char* name, const char* base_path, const char* file_name, uint8_t* data, size_t data_size) {
+  (void)name;
+  (void)base_path;
+  (void)file_name;
+  (void)data;
+  (void)data_size;
+  return false;
+}
+
+static retro_video_refresh_t video_refresh_cb = NULL;
+static retro_input_poll_t input_poll_cb = NULL;
+static retro_input_state_t input_state_cb = NULL;
+static retro_audio_sample_t audio_sample_cb = NULL;
+static retro_environment_t env_cb = NULL;
+
+sb_emu_state_t emu_state;
+
+/* ------------------ RETROARCH GB ----------------- */
+
+static gb_scratch_t gb_scratch;
```

Comment:
You should union the cores/scratch space for all the systems like the main frontend to save RAM.